test/e2e/deployment.go

```diff
@@ -374,28 +353,62 @@ func (d *Deployment) WaitForEnvoyUpdated() error {
 }
 
 func (d *Deployment) waitForEnvoyDaemonSetUpdated() error {
-	daemonSetUpdated := func() (bool, error) {
-		tempDS := new(apps_v1.DaemonSet)
-		if err := d.client.Get(context.TODO(), client.ObjectKeyFromObject(d.EnvoyDaemonSet), tempDS); err != nil {
+	labelSelectAppEnvoy := labels.SelectorFromSet(d.EnvoyDaemonSet.Spec.Selector.MatchLabels)
+	updatedPods := func() (bool, error) {
+		ds := &apps_v1.DaemonSet{}
+		if err := d.client.Get(context.TODO(), types.NamespacedName{Name: d.EnvoyDaemonSet.Name, Namespace: d.EnvoyDaemonSet.Namespace}, ds); err != nil {
 			return false, err
 		}
-		return tempDS.Status.ObservedGeneration == tempDS.Generation &&
-			tempDS.Status.NumberAvailable > 0 &&
-			tempDS.Status.NumberAvailable == tempDS.Status.DesiredNumberScheduled &&
-			tempDS.Status.UpdatedNumberScheduled == tempDS.Status.DesiredNumberScheduled, nil
+		updatedPods := int(ds.Status.DesiredNumberScheduled)
+		if len(ds.Spec.Template.Spec.Containers) > 1 {
+			updatedPods = d.getUpdatedPods(labelSelectAppEnvoy, d.EnvoyDaemonSet.Namespace)
+		}
+		return updatedPods == int(ds.Status.DesiredNumberScheduled) &&
+			ds.Status.NumberReady > 0, nil
 	}
-	return wait.PollImmediate(time.Millisecond*50, time.Minute*3, daemonSetUpdated)
+	return wait.PollImmediate(time.Millisecond*50, time.Minute*3, updatedPods)
 }
 
 func (d *Deployment) waitForEnvoyDeploymentUpdated() error {
-	deploymentUpdated := func() (bool, error) {
-		tempDeploy := new(apps_v1.Deployment)
-		if err := d.client.Get(context.TODO(), client.ObjectKeyFromObject(d.EnvoyDeployment), tempDeploy); err != nil {
-			return false, err
+	labelSelectAppEnvoy := labels.SelectorFromSet(d.EnvoyDeployment.Spec.Selector.MatchLabels)
+	updatedPods := func() (bool, error) {
+		updatedPods := d.getUpdatedPods(labelSelectAppEnvoy, d.EnvoyDeployment.Namespace)
+		return updatedPods == int(*d.EnvoyDeployment.Spec.Replicas) &&
```

Comment:
I'm not seeing exactly what the linter is complaining about here; `// nolint(gocritic)` is always an option if this is a false positive.

Reply:
Maybe my last push will do it, changed the order of the operands.